src/compressor/zstd/ZstdCompressor.h

```diff
@@ -22,15 +22,14 @@
 #include "include/encoding.h"
 #include "compressor/Compressor.h"
 
-#define COMPRESSION_LEVEL 5
-
 class ZstdCompressor : public Compressor {
+  CephContext *const cct;
```

Comment:
could you move the definition to the end of the class? see https://google.github.io/styleguide/cppguide.html#Declaration_Order

and i'd suggest use the just-enough information -- to keep `compression_level` instead of keeping a CephContext.
like
```c++
const int compression_level;
```

Reply:
I don't do a lot of C++ programming, so I based my changes off the other compressors.  This is how it was done in src/compressor/zlib/ZlibCompressor.h, so I essentially copied it.  I'll give your suggestions a shot though, but we'll want to modify the zlib compressor as well for consistency.

Reply:
Could you provide some more details on this one for someone with a mainly SRE background with some C and python programming experience?

Reply:
https://github.com/bstillwell/ceph/pull/1

Reply:
Thanks Kefu, but that patch isn't working correctly for me.  For one thing I can't change the compression level on the fly with:

`ceph daemon client.radosgw.$(hostname -s).422634.94342353273464 config set compressor_zstd_level 5`

It still compresses at level 1 after making that change.

It also no longer shows up as a configuration option:

`# ceph config ls | grep -i compressor
compressor_zlib_isal
compressor_zlib_level
qat_compressor_enabled`

Reply:
ahh, i didn't realize that you wanted to change the compression level on the fly. if that's the case the simplest way is surely to add that option to `src/common/legacy_config_opts.h` .

Reply:
> could you move the definition to the end of the class? 

so i only have a single request. see https://github.com/bstillwell/ceph/pull/1/files#diff-2ef96bb128fdd5ebc401b6cbb39e87efR104-R105 and https://github.com/bstillwell/ceph/pull/1/files#r409314118

Reply:
I believe this has now been taken care of.